.github/renovate.json

```diff
@@ -0,0 +1,33 @@
+{
```

Comment:
Does this file need to be in solr repository for the entire integration to work? Can't it be added to the folder after a clone by the GHA by the https://github.com/solrbot/renovate-github-action project?

Reply:
If it can live elsewhere easily then cool.  But I think it's inclusion isn't a blocker either.  I believe Ishan's concern is an appearance of "official"-ness?  Easily addressed with comments... but not sure in JSON.  Hmm.  A renovate-README.txt in this dir could have some commentary.

Reply:
I'll add a `dev-docs/dependency-upgrades.adoc` which both explains manual dep upgrades as well as documents what `.github/renovate.json` is for.

While tecnically we could perhaps embed all the config from `renovate.json` into the runner project, I like the fact that `renovate.json` gives the project a place to disable/enable the bot, tune schedules and handle future custom rules for dependencies that may require it.

Reply:
Please review the new dev-doc. It has some overlap with `help/dependencies.txt` but from different angles.

Reply:
> While tecnically we could perhaps embed all the config from renovate.json into the runner project

+1 to renovate.json staying in the runner project. Dependabot didn't need any custom config files in the main Solr project, AFAIK. I don't see a need for us to maintain configurations for a third party integration in our main repository.

Reply:
Thanks for the input. However, I disagree, and choose to keep it. It's analoguys to our `.lift.toml` and similar files used to enable and configure various tools.